From: Brendan Jackman <jackmanb@google.com>
Subject: x86/sev: Disable GCOV on noinstr object
Date: Mon, 15 Dec 2025 10:12:41 +0000

With Debian clang version 19.1.7 (3+build5) there are calls to
kasan_check_write() from __sev_es_nmi_complete, which violates noinstr. 
Fix it by disabling GCOV for the noinstr object, as has been done for
previous such instrumentation issues.

Note that this file already disables __SANITIZE_ADDRESS__ and
__SANITIZE_THREAD__, thus calls like kasan_check_write() ought to be nops
regardless of GCOV.  This has been fixed in other patches.  However, to
avoid any other accidental instrumentation showing up, (and since, in
principle GCOV is instrumentation and hence should be disabled for noinstr
code anyway), disable GCOV overall as well.

Link: https://lkml.kernel.org/r/20251215-gcov-inline-noinstr-v2-3-6f100b94fa99@google.com
Signed-off-by: Brendan Jackman <jackmanb@google.com>
Acked-by: Marco Elver <elver@google.com>
Cc: Alexander Potapenko <glider@google.com>
Cc: Andrey Konovalov <andreyknvl@gmail.com>
Cc: Andrey Ryabinin <ryabinin.a.a@gmail.com>
Cc: Ard Biesheuvel <ardb@kernel.org>
Cc: Dmitriy Vyukov <dvyukov@google.com>
Cc: Vincenzo Frascino <vincenzo.frascino@arm.com>
Cc: Kees Cook <keescook@chromium.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 arch/x86/coco/sev/Makefile |    2 ++
 1 file changed, 2 insertions(+)

--- a/arch/x86/coco/sev/Makefile~x86-sev-disable-gcov-on-noinstr-object
+++ a/arch/x86/coco/sev/Makefile
@@ -8,3 +8,5 @@ UBSAN_SANITIZE_noinstr.o	:= n
 # GCC may fail to respect __no_sanitize_address or __no_kcsan when inlining
 KASAN_SANITIZE_noinstr.o	:= n
 KCSAN_SANITIZE_noinstr.o	:= n
+
+GCOV_PROFILE_noinstr.o		:= n
_
